CoroutineTemplate/build.gradle

```diff
@@ -51,13 +50,12 @@ buildscript {
 }
 
 plugins {
-    id "io.gitlab.arturbosch.detekt" version "1.14.0"
+    id "io.gitlab.arturbosch.detekt" version "1.18.1"
```

Comment:
there seem to be quite many unrelated changes to the actual intention of this PR: `Refactor architecture` - next time please scope the PR appropriately so it doesn't cost too much time of the other to review 🙇

Reply:
@sleepylee I see, I will change this kind of habit 🙏